std/math/ff.asm

```diff
@@ -0,0 +1,60 @@
+/// Inverts `x` in the finite field with modulus `modulus`.
+/// Assumes that `modulus` is prime, but does not check it.
+let inverse = |x, modulus|
+    if x <= 0 || x >= modulus {
+        std::check::panic("Tried to compute the inverse of zero, of a negative number or a number outside the field.")
+    } else {
+        // TODO this is written in a complicated way
+        // because we do not have tuple destructuring assignment
+        (|r|
+            if r[0] < 0 { r[0] + modulus } else { r[0] }
+        )(extended_gcd(x, modulus))
+    };
+
+/// Computes `x + y` modulo the modulus.
+let add = |x, y, modulus| reduce(x + y, modulus);
+
+/// Computes `x - y` modulo the modulus.
+let sub = |x, y, modulus| reduce(x - y, modulus);
+
+/// Computes `x * y` modulo the modulus.
+/// TODO maybe just computing "(x * y) % modulus" is faster...
+let mul = |x, y, modulus|
+    match y {
+        0 => 0,
+        1 => x,
+        _ =>
+            add(
+                if y & 1 == 1 { x } else { 0 },
+                mul(x, y >> 1, modulus) << 1,
+                modulus
+            )
+    };
```

Comment:
```suggestion
let mul = |x, y, modulus| reduce(x * y, modulus);
```

Why not this?

Reply:
Because it creates an intermediate value that is twice as big. But as I said, probably it is still faster than running the "better" algorithm through the interpreter. Will test on big numbers.